diff --git a/build.mill b/build.mill
index 0c898366020..3499cd36e2f 100644
--- a/build.mill
+++ b/build.mill
@@ -252,12 +252,13 @@ object Deps {
   }
 }

-def millVersion: T[String] = Task { VcsVersion.vcsState().format() }
+def millVersion: T[String] = Task { "0.13.0-SCALA3" } // Task { VcsVersion.vcsState().format() }

 def millLastTag: T[String] = Task {
-  VcsVersion.vcsState().lastTag.getOrElse(
-    sys.error("No (last) git tag found. Your git history seems incomplete!")
-  )
+  "0.12.0-RC3"
+  // VcsVersion.vcsState().lastTag.getOrElse(
+  //   sys.error("No (last) git tag found. Your git history seems incomplete!")
+  // )
 }

 def millBinPlatform: T[String] = Task {
diff --git a/docs/package.mill b/docs/package.mill
index 3db8a792a7b..72489d50d56 100644
--- a/docs/package.mill
+++ b/docs/package.mill
@@ -170,7 +170,7 @@ object `package` extends RootModule {
     os.write.over(dest / "antora.yml", (lines ++ newLines).mkString("\n"))
   }

-  def githubPagesPlaybookText(authorMode: Boolean) = T.task { extraSources: Seq[os.Path] =>
+  def githubPagesPlaybookText(authorMode: Boolean) = Task.Anon { (extraSources: Seq[os.Path]) =>
     val taggedSources = for (path <- extraSources) yield {
       s"""    - url: ${build.baseDir}
          |      start_path: ${path.relativeTo(build.baseDir)}
@@ -232,23 +232,23 @@ object `package` extends RootModule {
   }

   def githubPages: T[PathRef] = Task {
-    generatePages(authorMode = false)().apply(oldDocSources().map(_.path))
+    generatePages(authorMode = false).apply().apply(oldDocSources().map(_.path))
   }

   def localPages = Task {
-    val pages = generatePages(authorMode = true)().apply(oldDocSources().map(_.path))
+    val pages = generatePages(authorMode = true).apply().apply(oldDocSources().map(_.path))
     T.log.outputStream.println(
       s"You can browse the local pages at: ${(pages.path / "index.html").toNIO.toUri()}"
     )
   }
   def fastPages = Task {
-    val pages = generatePages(authorMode = true)().apply(Nil)
+    val pages = generatePages(authorMode = true).apply().apply(Nil)
     T.log.outputStream.println(
       s"You can browse the local pages at: ${(pages.path / "index.html").toNIO.toUri()}"
     )
   }

-  def generatePages(authorMode: Boolean) = T.task { extraSources: Seq[os.Path] =>
+  def generatePages(authorMode: Boolean) = Task.Anon { (extraSources: Seq[os.Path]) =>
     T.log.errorStream.println("Creating Antora playbook ...")
     // dependency to sources
     source()
@@ -257,7 +257,7 @@ object `package` extends RootModule {
     val siteDir = docSite / "site"
     os.write(
       target = playbook,
-      data = githubPagesPlaybookText(authorMode)().apply(extraSources),
+      data = githubPagesPlaybookText(authorMode).apply().apply(extraSources),
       createFolders = true
     )
     T.log.errorStream.println("Running Antora ...")
diff --git a/mill-build/build.mill b/mill-build/build.mill
index 454349f6657..0fba6a9ccc9 100644
--- a/mill-build/build.mill
+++ b/mill-build/build.mill
@@ -4,12 +4,12 @@ import mill.scalalib._

 object `package` extends MillBuildRootModule {
   override def ivyDeps = Agg(
-    ivy"de.tototec::de.tobiasroeser.mill.vcs.version::0.4.0",
-    ivy"com.github.lolgab::mill-mima::0.1.1",
+    ivy"de.tototec::de.tobiasroeser.mill.vcs.version::0.4.0".withDottyCompat(scalaVersion()),
+    ivy"com.github.lolgab::mill-mima::0.1.1".withDottyCompat(scalaVersion()),
     ivy"net.sourceforge.htmlcleaner:htmlcleaner:2.29",
     // TODO: implement empty version for ivy deps as we do in import parser
     ivy"com.lihaoyi::mill-contrib-buildinfo:${mill.api.BuildInfo.millVersion}",
-    ivy"com.goyeau::mill-scalafix::0.4.1",
+    ivy"com.goyeau::mill-scalafix::0.4.1".withDottyCompat(scalaVersion()),
     ivy"com.lihaoyi::mill-main-graphviz:${mill.api.BuildInfo.millVersion}"
   )
 }
