diff --git a/mill-build/build.mill b/mill-build/build.mill
index 154569714a3..79609fb5233 100644
--- a/mill-build/build.mill
+++ b/mill-build/build.mill
@@ -10,7 +10,7 @@ object `package` extends MillBuildRootModule {
   override def mvnDeps = Seq(
     // Hardcode MIMA version so test-mill-bootstrap.sh works with
     // the locally-built `SNAPSHOT` version
-    mvn"com.github.lolgab::mill-mima_mill1.0.0-RC3:0.2.0-M4",
+//    mvn"com.github.lolgab::mill-mima_mill1.0.0-RC3:0.2.0-M4",
     mvn"net.sourceforge.htmlcleaner:htmlcleaner:2.29",
     // TODO: implement empty version for ivy deps as we do in import parser
     mvn"com.lihaoyi::mill-contrib-buildinfo:${mill.api.BuildInfo.millVersion}",
diff --git a/mill-build/src/millbuild/MillStableScalaModule.scala b/mill-build/src/millbuild/MillStableScalaModule.scala
index ef25e2a891f..11e78faf6da 100644
--- a/mill-build/src/millbuild/MillStableScalaModule.scala
+++ b/mill-build/src/millbuild/MillStableScalaModule.scala
@@ -1,15 +1,15 @@
 package millbuild
-import com.github.lolgab.mill.mima._
+//import com.github.lolgab.mill.mima._
 import mill._, scalalib._
 
 /** Publishable module which contains strictly handled API. */
-trait MillStableScalaModule extends MillPublishScalaModule with Mima {
+trait MillStableScalaModule extends MillPublishScalaModule /*with Mima*/ {
 
-  override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq.empty[ProblemFilter]
-
-  def mimaPreviousVersions: T[Seq[String]] = Settings.mimaBaseVersions
-
-  def mimaPreviousArtifacts: T[Seq[Dep]] = Task {
+//  override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq.empty[ProblemFilter]
+//
+//  def mimaPreviousVersions: T[Seq[String]] = Settings.mimaBaseVersions
+//
+//  def mimaPreviousArtifacts: T[Seq[Dep]] = Task {
 //    Settings.mimaBaseVersions
 //      .map({ version =>
 //        val patchedSuffix = {
@@ -27,8 +27,8 @@ trait MillStableScalaModule extends MillPublishScalaModule with Mima {
 //        val patchedId = artifactName() + patchedSuffix
 //        mvn"${pomSettings().organization}:${patchedId}:${version}"
 //      })
-    Seq.empty[Dep]
-  }
-
-  def mimaExcludeAnnotations = Seq("mill.api.internal.internal", "mill.api.experimental")
+//    Seq.empty[Dep]
+//  }
+//
+//  def mimaExcludeAnnotations = Seq("mill.api.internal.internal", "mill.api.experimental")
 }
