diff --git a/integration/package.mill b/integration/package.mill
index 5c388fafb6e..31763326edf 100644
--- a/integration/package.mill
+++ b/integration/package.mill
@@ -75,7 +75,8 @@ object `package` extends mill.Module {
             testReportXml(),
             javaHome().map(_.path),
             testParallelism(),
-            testLogLevel()
+            testLogLevel(),
+            jvmWorker().worker()
           )
           testModuleUtil.runTests()
         }
