diff --git a/build.mill b/build.mill
index d815508132a..ba62e147383 100644
--- a/build.mill
+++ b/build.mill
@@ -19,9 +19,8 @@ import mill.define.Cross
 import scala.util.Properties
 
 // plugins and dependencies
-import $meta._
+//| meta: true
 
-import $packages._
 
 object Settings {
   val pomOrg = "com.lihaoyi"
diff --git a/mill-build/build.mill b/mill-build/build.mill
index f32b3cbcf65..3fc9b0097c8 100644
--- a/mill-build/build.mill
+++ b/mill-build/build.mill
@@ -14,7 +14,7 @@ object `package` extends MillBuildRootModule {
     // TODO: implement empty version for ivy deps as we do in import parser
     mvn"com.lihaoyi::mill-contrib-buildinfo:${mill.api.BuildInfo.millVersion}",
 //    mvn"com.goyeau::mill-scalafix::0.5.0",
-    mvn"com.lihaoyi::mill-main-graphviz:${mill.api.BuildInfo.millVersion}",
+    mvn"com.lihaoyi::mill-libs-graphviz:${mill.api.BuildInfo.millVersion}",
     // TODO: document, why we have this dependency
     mvn"org.jsoup:jsoup:1.18.1"
   )
