diff --git a/.github/workflows/autofix.yml b/.github/workflows/autofix.yml
index fff08f9d91e..4b869becf2c 100644
--- a/.github/workflows/autofix.yml
+++ b/.github/workflows/autofix.yml
@@ -10,7 +10,6 @@ jobs:
       - uses: sbt/setup-sbt@v1
 
       - run: |
-          ./mill __.fix + mill.javalib.palantirformat.PalantirFormatModule/ + mill.scalalib.scalafmt.ScalafmtModule/ + mill.kotlinlib.ktlint.KtlintModule/
-          ./mill --meta-level 1 mill.scalalib.scalafmt.ScalafmtModule/
+          ./mill __.fix + mill.javalib.palantirformat.PalantirFormatModule/ + mill.scalalib.scalafmt.ScalafmtModule/scalafmt + mill.kotlinlib.ktlint.KtlintModule/
 
       - uses: autofix-ci/action@551dded8c6cc8a1054039c8bc0b8b48c51dfc6ef
\ No newline at end of file
diff --git a/.github/workflows/run-tests.yml b/.github/workflows/run-tests.yml
index b3d16508f21..01afa62e23c 100644
--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -305,5 +305,4 @@ jobs:
       java-version: '17'
       buildcmd: |
         set -eux
-        ./mill -i mill.scalalib.scalafmt.ScalafmtModule/checkFormatAll + __.mimaReportBinaryIssues + __.fix --check + mill.javalib.palantirformat.PalantirFormatModule/ --check + mill.kotlinlib.ktlint.KtlintModule/checkFormatAll
-        ./mill -i --meta-level 1 mill.scalalib.scalafmt.ScalafmtModule/checkFormatAll
+        ./mill -i mill.scalalib.scalafmt.ScalafmtModule/scalafmt --check + __.mimaReportBinaryIssues + __.fix --check + mill.javalib.palantirformat.PalantirFormatModule/ --check + mill.kotlinlib.ktlint.KtlintModule/checkFormatAll
diff --git a/build.mill b/build.mill
index 5a46cd47946..b016ae330af 100644
--- a/build.mill
+++ b/build.mill
@@ -1,15 +1,15 @@
 package build
 // imports
-import com.github.lolgab.mill.mima.Mima
+//import com.github.lolgab.mill.mima.Mima
 import coursier.maven.MavenRepository
 import coursier.VersionConstraint
-import de.tobiasroeser.mill.vcs.version.VcsVersion
-import com.goyeau.mill.scalafix.ScalafixModule
+import mill.main.VcsVersion
+//import com.goyeau.mill.scalafix.ScalafixModule
 import mill._
 import mill.define.NamedTask
 import mill.main.Tasks
 import mill.scalalib._
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.define.SelectMode
 import mill.T
@@ -171,12 +171,12 @@ object Deps {
   val testng = ivy"org.testng:testng:7.5.1"
   val sbtTestInterface = ivy"org.scala-sbt:test-interface:1.0"
   def scalaCompiler(scalaVersion: String) = {
-    if (ZincWorkerUtil.isScala3(scalaVersion)) ivy"org.scala-lang:scala3-compiler_3:${scalaVersion}"
+    if (JvmWorkerUtil.isScala3(scalaVersion)) ivy"org.scala-lang:scala3-compiler_3:${scalaVersion}"
     else ivy"org.scala-lang:scala-compiler:${scalaVersion}"
   }
   val scalafmtDynamic = ivy"org.scalameta::scalafmt-dynamic:3.8.5".withDottyCompat(scalaVersion)
   def scalaReflect(scalaVersion: String) =
-    if (ZincWorkerUtil.isScala3(scalaVersion))
+    if (JvmWorkerUtil.isScala3(scalaVersion))
       ivy"org.scala-lang:scala-reflect:${Deps.scala2Version}"
     else ivy"org.scala-lang:scala-reflect:${scalaVersion}"
   val scoverage2Version = "2.2.1"
@@ -475,13 +475,13 @@ trait MillPublishJavaModule extends MillJavaModule with PublishModule {
 /**
  * Some custom scala settings and test convenience
  */
-trait MillScalaModule extends ScalaModule with MillJavaModule with ScalafixModule { outer =>
+trait MillScalaModule extends ScalaModule with MillJavaModule/* with ScalafixModule*/ { outer =>
   def scalaVersion = Deps.scalaVersion
   def scalapVersion: T[String] = Deps.scala2Version
   def scalafixScalaBinaryVersion = T {
     def sv = scalaVersion()
-    if (ZincWorkerUtil.isScala3(sv)) "2.13"
-    else ZincWorkerUtil.scalaBinaryVersion(sv)
+    if (JvmWorkerUtil.isScala3(sv)) "2.13"
+    else JvmWorkerUtil.scalaBinaryVersion(sv)
   }
 
   def scalafixConfig = T { Some(T.workspace / ".scalafix.conf") }
@@ -495,7 +495,7 @@ trait MillScalaModule extends ScalaModule with MillJavaModule with ScalafixModul
       "-deprecation",
       "-feature"
     ) ++ (
-      if (ZincWorkerUtil.isScala3(scalaVersion())) Seq(
+      if (JvmWorkerUtil.isScala3(scalaVersion())) Seq(
         // "-Werror",
         "-Wunused:all",
         // "-Xfatal-warnings",
@@ -529,25 +529,25 @@ trait MillScalaModule extends ScalaModule with MillJavaModule with ScalafixModul
 
   def scalacPluginIvyDeps = T {
     val sv = scalaVersion()
-    val binaryVersion = ZincWorkerUtil.scalaBinaryVersion(sv)
+    val binaryVersion = JvmWorkerUtil.scalaBinaryVersion(sv)
     val hasModuleDefs = binaryVersion == "2.13" || binaryVersion == "3"
     super.scalacPluginIvyDeps() ++
-      Agg.when(binaryVersion != "3")(Deps.acyclic) ++
-      Agg.when(hasModuleDefs)(Deps.millModuledefsPlugin)
+      Option.when(binaryVersion != "3")(Deps.acyclic) ++
+      Option.when(hasModuleDefs)(Deps.millModuledefsPlugin)
   }
 
   def mandatoryIvyDeps = T {
     val sv = scalaVersion()
-    val binaryVersion = ZincWorkerUtil.scalaBinaryVersion(sv)
+    val binaryVersion = JvmWorkerUtil.scalaBinaryVersion(sv)
     val hasModuleDefs = binaryVersion == "2.13" || binaryVersion == "3"
     super.mandatoryIvyDeps() ++
-      Agg.when(hasModuleDefs)(Deps.millModuledefs)
+      Option.when(hasModuleDefs)(Deps.millModuledefs)
   }
 
   /** Default tests module. */
   lazy val test: MillScalaTests = new MillScalaTests {}
   trait MillScalaTests extends ScalaTests with MillJavaModule with MillBaseTestsModule
-      with ScalafixModule {
+      /*with ScalafixModule*/ {
     def scalafixConfig = T { Some(T.workspace / ".scalafix.conf") }
     def forkArgs = super.forkArgs() ++ outer.testArgs()
     def moduleDeps = outer.testModuleDeps
@@ -589,7 +589,8 @@ trait MillBaseTestsModule extends TestModule {
 trait MillPublishScalaModule extends MillScalaModule with MillPublishJavaModule
 
 /** Publishable module which contains strictly handled API. */
-trait MillStableScalaModule extends MillPublishScalaModule with Mima {
+trait MillStableScalaModule extends MillPublishScalaModule /*with Mima*/ {
+  /*
   import com.github.lolgab.mill.mima._
 //  override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq()
 
@@ -623,7 +624,7 @@ trait MillStableScalaModule extends MillPublishScalaModule with Mima {
   def skipPreviousVersions: T[Seq[String]] = T {
     T.log.info("Skipping mima for previous versions (!!1000s of errors due to Scala 3)")
     mimaPreviousVersions() // T(Seq.empty[String])
-  }
+  }*/
 }
 
 trait MillPublishCrossScalaModule extends MillPublishJavaModule with CrossScalaModule
@@ -639,7 +640,7 @@ trait BridgeModule extends MillPublishCrossScalaModule {
     ivy"org.scala-sbt:compiler-interface:${Deps.zinc.version}",
     ivy"org.scala-sbt:util-interface:${Deps.zinc.version}"
   ) ++ Agg(
-    if (ZincWorkerUtil.isScala3(crossScalaVersion))
+    if (JvmWorkerUtil.isScala3(crossScalaVersion))
       ivy"org.scala-lang::scala3-compiler:${crossScalaVersion}"
     else ivy"org.scala-lang:scala-compiler:${crossScalaVersion}"
   )
@@ -650,7 +651,7 @@ trait BridgeModule extends MillPublishCrossScalaModule {
   }
 
   def compilerBridgeIvyDeps: T[Agg[Dep]] = Agg(
-    (if (ZincWorkerUtil.isScala3(crossScalaVersion))
+    (if (JvmWorkerUtil.isScala3(crossScalaVersion))
        ivy"org.scala-lang:scala3-sbt-bridge:${crossScalaVersion}"
      else ivy"org.scala-sbt::compiler-bridge:${Deps.zinc.version}").exclude("*" -> "*")
   )
diff --git a/contrib/package.mill b/contrib/package.mill
index 5100b411e15..0b7b1ba3440 100644
--- a/contrib/package.mill
+++ b/contrib/package.mill
@@ -5,7 +5,7 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.main.Tasks
 import mill.scalalib._
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.define.SelectMode
@@ -28,7 +28,7 @@ import $meta._
  */
 object `package` extends RootModule {
   def contribModules: Seq[ContribModule] =
-    moduleInternal.modules.collect { case m: ContribModule => m }
+    moduleInternal.modules.collect { case m: ContribModule => m }
 
   trait ContribModule extends build.MillPublishScalaModule {
     def readme = Task.Source("readme.adoc")
@@ -84,9 +84,9 @@ object `package` extends RootModule {
     object worker extends Cross[WorkerModule](build.Deps.play.keys.toSeq)
     trait WorkerModule extends build.MillPublishScalaModule with Cross.Module[String] {
       def playBinary = crossValue
-      def millSourcePath: os.Path = super.millSourcePath / playBinary
+      def moduleDir: os.Path = super.moduleDir / playBinary
 
-      def sharedSources = Task.Sources(moduleDir / os.up / "src-shared")
+      def sharedSources = Task.Sources(moduleDir / os.up / "src-shared")
       def sources = Task {
         // We want to avoid duplicating code as long as the Play APIs allow.
         // But if newer Play versions introduce incompatibilities,
@@ -139,7 +139,7 @@ object `package` extends RootModule {
           build.Deps.scalacScoverage2Reporter,
           build.Deps.scalacScoverage2Domain,
           build.Deps.scalacScoverage2Serializer
-        ) ++ Agg.when(!ZincWorkerUtil.isScala3(scalaVersion()))(build.Deps.scalacScoverage2Plugin)
+        ) ++ Option.when(!JvmWorkerUtil.isScala3(scalaVersion()))(build.Deps.scalacScoverage2Plugin)
       }
       def mandatoryIvyDeps = Agg.empty[Dep]
     }
diff --git a/core/codesig/package.mill b/core/codesig/package.mill
index e49f218f4be..9d03aa073f3 100644
--- a/core/codesig/package.mill
+++ b/core/codesig/package.mill
@@ -18,8 +18,8 @@ object `package` extends RootModule with build.MillPublishScalaModule {
   override lazy val test: CodeSigTests = new CodeSigTests {}
   trait CodeSigTests extends MillScalaTests {
     val caseKeys = build.interp.watchValue(
-      os.walk(moduleDir / "cases", maxDepth = 3)
-        .map(_.subRelativeTo(moduleDir / "cases").segments)
+      os.walk(moduleDir / "cases", maxDepth = 3)
+        .map(_.subRelativeTo(moduleDir / "cases").segments)
         .collect { case Seq(a, b, c) => s"$a-$b-$c" }
     )
 
@@ -45,7 +45,7 @@ object `package` extends RootModule with build.MillPublishScalaModule {
       def moduleDeps = Seq(external)
 
       val Array(prefix, suffix, rest) = caseName.split("-", 3)
-      def millSourcePath = super.millSourcePath / prefix / suffix / rest
+      def moduleDir = super.moduleDir / prefix / suffix / rest
       def scalaVersion = build.Deps.scalaVersion
       def ivyDeps = Task {
         if (!caseName.contains("realistic") && !caseName.contains("sourcecode")) super.ivyDeps()
diff --git a/dist/package.mill b/dist/package.mill
index 488648f42a0..9d9d3fcf73c 100644
--- a/dist/package.mill
+++ b/dist/package.mill
@@ -2,14 +2,14 @@ package build.dist
 import mill._, scalalib._, publish._
 import mill.define.ModuleRef
 import mill.util.Jvm
-import de.tobiasroeser.mill.vcs.version.VcsVersion
+//import de.tobiasroeser.mill.vcs.version.VcsVersion
 
 import scala.util.Using
 
 trait InstallModule extends build.MillPublishJavaModule {
   // All modules that we want to aggregate as part of this `dev` assembly.
   // Excluding itself, and the `dist` module that uses it
-  lazy val allPublishModules = build.moduleInternal.modules.collect {
+  lazy val allPublishModules = build.moduleInternal.modules.collect {
     case m: PublishModule if (m ne build.dist) && (m ne build.dist.native) => m
   }
   def moduleDeps = Seq(build.runner, build.idea, build.main.init)
@@ -45,7 +45,7 @@ trait InstallModule extends build.MillPublishJavaModule {
         (os.home / ".cache/mill/download" / (build.millVersion() + cacheBinarySuffix + batExt)).toString()
       )
     )()
-    Task.log.outputStream.println(path.toString())
+    Task.log.streams.out.println(path.toString())
     PathRef(path)
   }
 
@@ -235,7 +235,7 @@ object `package` extends RootModule with InstallModule {
   def examplePathsWithArtifactName: Task[Seq[(os.Path, String)]] = Task.Anon {
     for {
       exampleMod <- build.example.exampleModules
-      path = exampleMod.moduleDir
+      path = exampleMod.moduleDir
     } yield {
       val example = path.subRelativeTo(Task.workspace)
       val artifactName = example.segments.mkString("-")
@@ -267,6 +267,7 @@ object `package` extends RootModule with InstallModule {
   }
 
   def uploadToGithub(authKey: String) = Task.Command {
+    /*
     val vcsState = VcsVersion.vcsState()
     val label = vcsState.copy(dirtyHash = None).format()
     if (label != build.millVersion()) sys.error("Modified mill version detected, aborting upload")
@@ -283,7 +284,7 @@ object `package` extends RootModule with InstallModule {
         headers = Seq("Authorization" -> ("token " + authKey))
       )
     }
-
+    */
     ()
   }
 
@@ -322,16 +323,16 @@ object `package` extends RootModule with InstallModule {
         out.write(os.read.bytes(assembly().path))
       }
 
-      if (!mill.main.client.Util.isWindows) os.perms.set(executable, "rwxrwxrwx")
+      if (!mill.constants.Util.isWindows) os.perms.set(executable, "rwxrwxrwx")
 
       PathRef(executable)
     }
 
     def nativeImageOptions = Seq("--no-fallback", "--enable-url-protocols=https", "-Os")
 
-    def zincWorker = ModuleRef(ZincWorkerGraalvm)
+    def jvmWorker = ModuleRef(JvmWorkerGraalvm)
 
-    object ZincWorkerGraalvm extends ZincWorkerModule {
+    object JvmWorkerGraalvm extends JvmWorkerModule {
       def jvmId = build.Settings.graalvmJvmId
     }
   }
diff --git a/example/package.mill b/example/package.mill
index c1f697ea643..83475756e83 100644
--- a/example/package.mill
+++ b/example/package.mill
@@ -5,7 +5,7 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.main.Tasks
 import mill.scalalib._
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.define.SelectMode
@@ -17,117 +17,117 @@ import mill.define.Cross
 import $meta._
 
 object `package` extends RootModule with Module {
-  def exampleModules: Seq[ExampleCrossModule] = moduleInternal
+  def exampleModules: Seq[ExampleCrossModule] = moduleInternal
     .modules
     .collect { case m: ExampleCrossModule => m }
 
   object android extends Module {
     object javalib
-        extends Cross[ExampleCrossModuleAndroid](build.listIn(moduleDir / "javalib"))
+        extends Cross[ExampleCrossModuleAndroid](build.listIn(moduleDir / "javalib"))
     object kotlinlib
-        extends Cross[ExampleCrossModuleAndroid](build.listIn(moduleDir / "kotlinlib"))
+        extends Cross[ExampleCrossModuleAndroid](build.listIn(moduleDir / "kotlinlib"))
   }
   object javalib extends Module {
 
-    object basic extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "basic"))
-    object module extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "module"))
+    object basic extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "basic"))
+    object module extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "module"))
     object dependencies
-        extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "dependencies"))
-    object testing extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "testing"))
-    object linting extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "linting"))
+        extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "dependencies"))
+    object testing extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "testing"))
+    object linting extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "linting"))
     object publishing
-        extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "publishing"))
-    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
+        extends Cross[ExampleCrossModuleJava](build.listIn(moduleDir / "publishing"))
+    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
   }
   object kotlinlib extends Module {
-    object basic extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "basic"))
-    object module extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "module"))
+    object basic extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "basic"))
+    object module extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "module"))
     object dependencies
-        extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "dependencies"))
-    object testing extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "testing"))
-    object linting extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "linting"))
+        extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "dependencies"))
+    object testing extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "testing"))
+    object linting extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "linting"))
     object publishing
-        extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "publishing"))
-    object web extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "web"))
+        extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "publishing"))
+    object web extends Cross[ExampleCrossModuleKotlin](build.listIn(moduleDir / "web"))
   }
   object scalalib extends Module {
-    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
-    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
+    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
+    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
     object dependencies
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
-    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
-    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
-    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
-    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
-    object native extends Cross[ExampleCrossModule](build.listIn(moduleDir / "native"))
-    object spark extends Cross[ExampleCrossModule](build.listIn(moduleDir / "spark"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
+    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
+    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
+    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
+    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
+    object native extends Cross[ExampleCrossModule](build.listIn(moduleDir / "native"))
+    object spark extends Cross[ExampleCrossModule](build.listIn(moduleDir / "spark"))
   }
 
   object migrating extends Module {
     object javalib
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "javalib"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "javalib"))
     object scalalib
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "scalalib"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "scalalib"))
   }
   object javascriptlib extends Module {
-    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
-    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
-    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
+    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
+    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
+    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
     object dependencies
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
-    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
-    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
+    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
+    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
   }
   object pythonlib extends Module {
-    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
+    object basic extends Cross[ExampleCrossModule](build.listIn(moduleDir / "basic"))
     object dependencies
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
-    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
-    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
-    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
-    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
-    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
+    object linting extends Cross[ExampleCrossModule](build.listIn(moduleDir / "linting"))
+    object publishing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "publishing"))
+    object module extends Cross[ExampleCrossModule](build.listIn(moduleDir / "module"))
+    object web extends Cross[ExampleCrossModule](build.listIn(moduleDir / "web"))
+    object testing extends Cross[ExampleCrossModule](build.listIn(moduleDir / "testing"))
   }
 
   object cli extends Module {
-    object builtins extends Cross[ExampleCrossModule](build.listIn(moduleDir / "builtins"))
+    object builtins extends Cross[ExampleCrossModule](build.listIn(moduleDir / "builtins"))
   }
 
   object fundamentals extends Module {
 
     object dependencies
-        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
-    object tasks extends Cross[ExampleCrossModule](build.listIn(moduleDir / "tasks"))
-    object modules extends Cross[ExampleCrossModule](build.listIn(moduleDir / "modules"))
-    object cross extends Cross[ExampleCrossModule](build.listIn(moduleDir / "cross"))
-    object `out-dir` extends Cross[ExampleCrossModule](build.listIn(moduleDir / "out-dir"))
-    object libraries extends Cross[ExampleCrossModule](build.listIn(moduleDir / "libraries"))
+        extends Cross[ExampleCrossModule](build.listIn(moduleDir / "dependencies"))
+    object tasks extends Cross[ExampleCrossModule](build.listIn(moduleDir / "tasks"))
+    object modules extends Cross[ExampleCrossModule](build.listIn(moduleDir / "modules"))
+    object cross extends Cross[ExampleCrossModule](build.listIn(moduleDir / "cross"))
+    object `out-dir` extends Cross[ExampleCrossModule](build.listIn(moduleDir / "out-dir"))
+    object libraries extends Cross[ExampleCrossModule](build.listIn(moduleDir / "libraries"))
   }
 
   object depth extends Module {
 
-    object sandbox extends Cross[ExampleCrossModule](build.listIn(moduleDir / "sandbox"))
-    object javahome extends Cross[ExampleCrossModule](build.listIn(moduleDir / "javahome"))
+    object sandbox extends Cross[ExampleCrossModule](build.listIn(moduleDir / "sandbox"))
+    object javahome extends Cross[ExampleCrossModule](build.listIn(moduleDir / "javahome"))
   }
   object large extends Module {
 
-    object selective extends Cross[ExampleCrossModule](build.listIn(moduleDir / "selective"))
-    object multifile extends Cross[ExampleCrossModule](build.listIn(moduleDir / "multifile"))
-    object multilang extends Cross[ExampleCrossModule](build.listIn(moduleDir / "multilang"))
+    object selective extends Cross[ExampleCrossModule](build.listIn(moduleDir / "selective"))
+    object multifile extends Cross[ExampleCrossModule](build.listIn(moduleDir / "multifile"))
+    object multilang extends Cross[ExampleCrossModule](build.listIn(moduleDir / "multilang"))
   }
 
   object extending extends Module {
-    object imports extends Cross[ExampleCrossModule](build.listIn(moduleDir / "imports"))
-    object metabuild extends Cross[ExampleCrossModule](build.listIn(moduleDir / "metabuild"))
-    object plugins extends Cross[ExampleCrossModule](build.listIn(moduleDir / "plugins"))
-    object jvmcode extends Cross[ExampleCrossModule](build.listIn(moduleDir / "jvmcode"))
-    object python extends Cross[ExampleCrossModule](build.listIn(moduleDir / "python"))
-    object typescript extends Cross[ExampleCrossModule](build.listIn(moduleDir / "typescript"))
+    object imports extends Cross[ExampleCrossModule](build.listIn(moduleDir / "imports"))
+    object metabuild extends Cross[ExampleCrossModule](build.listIn(moduleDir / "metabuild"))
+    object plugins extends Cross[ExampleCrossModule](build.listIn(moduleDir / "plugins"))
+    object jvmcode extends Cross[ExampleCrossModule](build.listIn(moduleDir / "jvmcode"))
+    object python extends Cross[ExampleCrossModule](build.listIn(moduleDir / "python"))
+    object typescript extends Cross[ExampleCrossModule](build.listIn(moduleDir / "typescript"))
   }
 
   trait ExampleCrossModuleKotlin extends ExampleCrossModuleJava {
 
-    override def lineTransform(line: String) = this.moduleSegments.parts.last match {
+    override def lineTransform(line: String) = this.moduleSegments.parts.last match {
       case "1-test-suite" => line
           .replace(
             "mill bar.test bar.BarTests.hello",
@@ -153,7 +153,7 @@ object `package` extends RootModule with Module {
       case "publishing" => Some(scalalib.publishing)
       case _ => None
     }
-    val upstreamOpt = upstreamCross(this.moduleSegments.parts.dropRight(1).last)
+    val upstreamOpt = upstreamCross(this.moduleSegments.parts.dropRight(1).last)
       .flatMap(_.valuesToModules.get(List(crossValue)))
 
     def testRepoRoot = Task {
@@ -219,10 +219,10 @@ object `package` extends RootModule with Module {
 
   trait ExampleCrossModule extends build.integration.IntegrationTestModule {
     // disable scalafix because these example modules don't have sources causing it to misbehave
-    def testRepoSourceRoot: T[PathRef] = Task.Source(moduleDir)
+    def testRepoSourceRoot: T[PathRef] = Task.Source(moduleDir)
     def testRepoRoot: T[PathRef] = Task { testRepoSourceRoot() }
 
-    def sources0 = Task.Sources(moduleDir)
+    def sources0 = Task.Sources(moduleDir)
     def sources = Task {
       sources0()
         .flatMap(pathRef => os.walk(pathRef.path))
@@ -251,7 +251,7 @@ object `package` extends RootModule with Module {
 
     def rendered = Task {
       var seenCode = false
-      val examplePath = moduleDir.subRelativeTo(Task.workspace)
+      val examplePath = moduleDir.subRelativeTo(Task.workspace)
       os.write(
         Task.dest / "example.adoc",
         parsed()
@@ -310,7 +310,7 @@ $txt
     "arrow" -> ("arrow-kt/arrow", "bc9bf92cc98e01c21bdd2bf8640cf7db0f97204a"),
     "ollama-js" -> ("ollama/ollama-js", "99293abe2c7c27ce7e76e8b4a98cae948f00058d")
   )
-  object thirdparty extends Cross[ThirdPartyModule](build.listIn(moduleDir / "thirdparty"))
+  object thirdparty extends Cross[ThirdPartyModule](build.listIn(moduleDir / "thirdparty"))
   trait ThirdPartyModule extends ExampleCrossModule {
     val (repoPath, repoHash) = repoInfo(crossValue)
     def repoSlug = repoPath.split("/").last
diff --git a/integration/package.mill b/integration/package.mill
index 204b8c2e273..decd732463d 100644
--- a/integration/package.mill
+++ b/integration/package.mill
@@ -5,7 +5,7 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.main.Tasks
 import mill.scalalib._
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.define.SelectMode
@@ -23,7 +23,7 @@ object `package` extends RootModule {
   // and pass `lib`'s compile output back to `test`.
 
   trait IntegrationTestModule extends Cross.Module[String] {
-    def millSourcePath = super.millSourcePath / crossValue
+    def moduleDir = super.moduleDir / crossValue
 
     def moduleDeps = Seq(build.main.test, build.testkit, build.runner)
 
@@ -33,11 +33,11 @@ object `package` extends RootModule {
     def forkEnv: T[Map[String, String]]
     def testExclusive = false
     trait ModeModule extends build.MillBaseTestsModule {
-      override def test(args: String*) = Task.Command(exclusive = testExclusive) {
+      override def testForked(args: String*) = Task.Command(exclusive = testExclusive) {
         testTask(Task.Anon { args }, Task.Anon { Seq.empty[String] })()
       }
 
-      def mode: String = moduleSegments.parts.last
+      def mode: String = moduleSegments.parts.last
       def scalaVersion = build.Deps.scalaVersion
 
       def forkEnv =
@@ -97,15 +97,15 @@ object `package` extends RootModule {
     }
   }
 
-  object failure extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "failure"))
-  object feature extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "feature"))
+  object failure extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "failure"))
+  object feature extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "feature"))
   object invalidation
-      extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "invalidation"))
-  object ide extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "ide"))
+      extends Cross[IntegrationCrossModule](build.listIn(moduleDir / "invalidation"))
+  object ide extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "ide"))
   object bootstrap
-      extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "bootstrap"))
+      extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "bootstrap"))
   object migrating
-      extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "migrating"))
+      extends Cross[IdeIntegrationCrossModule](build.listIn(moduleDir / "migrating"))
   trait IntegrationCrossModule extends build.MillScalaModule with IntegrationTestModule {
     override def moduleDeps = super[IntegrationTestModule].moduleDeps
     def forkEnv = super.forkEnv() ++ Seq(
diff --git a/main/package.mill b/main/package.mill
index 93e12df4f03..a1d4a1c0132 100644
--- a/main/package.mill
+++ b/main/package.mill
@@ -5,7 +5,7 @@ import mill.scalalib._
 import mill.contrib.buildinfo.BuildInfo
 import mill.T
 import mill.define.Cross
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 
 /**
  * This package contains the helpers necessary to define the `build.*` builtin helpers,
@@ -30,7 +30,7 @@ object `package` extends RootModule with build.MillStableScalaModule with BuildI
   )
 
   def compileIvyDeps = Task {
-    if (ZincWorkerUtil.isScala3(scalaVersion())) Agg.empty
+    if (JvmWorkerUtil.isScala3(scalaVersion())) Agg.empty
     else Agg(build.Deps.scalaReflect(scalaVersion()))
   }
 
diff --git a/mill-build/build.mill b/mill-build/build.mill
index 6f6c07e7f85..521b6533dd6 100644
--- a/mill-build/build.mill
+++ b/mill-build/build.mill
@@ -8,12 +8,12 @@ object `package` extends MillBuildRootModule {
   def sources = super.sources() ++ sources0()
 
   override def ivyDeps = Agg(
-    ivy"de.tototec::de.tobiasroeser.mill.vcs.version::0.4.1",
-    ivy"com.github.lolgab::mill-mima::0.1.1",
+//    ivy"de.tototec::de.tobiasroeser.mill.vcs.version::0.4.1",
+//    ivy"com.github.lolgab::mill-mima::0.1.1",
     ivy"net.sourceforge.htmlcleaner:htmlcleaner:2.29",
     // TODO: implement empty version for ivy deps as we do in import parser
     ivy"com.lihaoyi::mill-contrib-buildinfo:${mill.api.BuildInfo.millVersion}",
-    ivy"com.goyeau::mill-scalafix::0.5.0",
+//    ivy"com.goyeau::mill-scalafix::0.5.0",
     ivy"com.lihaoyi::mill-main-graphviz:${mill.api.BuildInfo.millVersion}",
     // TODO: document, why we have this dependency
     ivy"org.jsoup:jsoup:1.18.1"
diff --git a/scalajslib/package.mill b/scalajslib/package.mill
index ee6663745f9..09b545ca7a5 100644
--- a/scalajslib/package.mill
+++ b/scalajslib/package.mill
@@ -44,7 +44,7 @@ object `package` extends RootModule with build.MillStableScalaModule with BuildI
   object worker extends Cross[WorkerModule]("1")
   trait WorkerModule extends build.MillPublishScalaModule with Cross.Module[String] {
     def scalajsWorkerVersion = crossValue
-    def millSourcePath: os.Path = super.millSourcePath / scalajsWorkerVersion
+    def moduleDir: os.Path = super.moduleDir / scalajsWorkerVersion
     def compileModuleDeps = Seq(build.scalajslib.`worker-api`, build.core.constants, build.core.api)
     def mandatoryIvyDeps = Agg.empty[Dep]
     def compileIvyDeps = super.mandatoryIvyDeps() ++ Agg(
diff --git a/scalalib/package.mill b/scalalib/package.mill
index 538aa270336..5bb77a9d3cf 100644
--- a/scalalib/package.mill
+++ b/scalalib/package.mill
@@ -6,7 +6,7 @@ import mill._
 import mill.define.NamedTask
 import mill.main.Tasks
 import mill.scalalib._
-import mill.scalalib.api.ZincWorkerUtil
+import mill.scalalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.define.SelectMode
diff --git a/scalanativelib/package.mill b/scalanativelib/package.mill
index a0c2c525a9e..141559d8435 100644
--- a/scalanativelib/package.mill
+++ b/scalanativelib/package.mill
@@ -17,7 +17,7 @@ object `package` extends RootModule with build.MillStableScalaModule {
 
   trait WorkerModule extends build.MillPublishScalaModule with Cross.Module[String] {
     def scalaNativeWorkerVersion = crossValue
-    def millSourcePath: os.Path = super.millSourcePath / scalaNativeWorkerVersion
+    def moduleDir: os.Path = super.moduleDir / scalaNativeWorkerVersion
     def compileModuleDeps = Seq(`worker-api`)
     def compileIvyDeps = scalaNativeWorkerVersion match {
       case "0.5" =>
diff --git a/website/package.mill b/website/package.mill
index a3ef4cc782d..81028a85cf4 100644
--- a/website/package.mill
+++ b/website/package.mill
@@ -26,7 +26,7 @@ object `package` extends RootModule {
         !ref.path.lastOpt.exists(_.startsWith("dependency_2.13-"))
       }
     def scalaVersion = build.Deps.scalaVersion
-    def moduleDeps = build.moduleInternal.modules.collect {
+    def moduleDeps = build.moduleInternal.modules.collect {
       case m: build.MillStableScalaModule => m
       case m: JavaModule if m eq build.kotlinlib => m
     }
@@ -84,7 +84,7 @@ object `package` extends RootModule {
     val renderedExamples: Seq[(os.SubPath, PathRef)] =
       Task.traverse(build.example.exampleModules)(m =>
         Task.Anon {
-          (m.moduleDir.subRelativeTo(build.example.moduleDir), m.rendered())
+          (m.moduleDir.subRelativeTo(build.example.moduleDir), m.rendered())
         }
       )()
 
@@ -96,7 +96,7 @@ object `package` extends RootModule {
 
     val contribReadmes = Task.traverse(build.contrib.contribModules)(m =>
       Task.Anon {
-        m.moduleSegments.parts.last -> m.readme()
+        m.moduleSegments.parts.last -> m.readme()
       }
     )()
 
@@ -335,21 +335,21 @@ object `package` extends RootModule {
 
   def localPages: T[PathRef] = Task {
     val pages = generatePages(authorMode = true).apply().apply(oldDocSources().map(_.path))
-    Task.log.outputStream.println(
+    Task.log.streams.out.println(
       s"You can browse the pages at: ${(pages.path / "index.html").toNIO.toUri()}"
     )
     pages
   }
   def fastPages: T[PathRef] = Task {
     val pages = generatePages(authorMode = true).apply().apply(Nil)
-    Task.log.outputStream.println(
+    Task.log.streams.out.println(
       s"You can browse the pages at: ${(pages.path / "index.html").toNIO.toUri()}"
     )
     pages
   }
 
   def generatePages(authorMode: Boolean) = Task.Anon { (extraSources: Seq[os.Path]) =>
-    Task.log.errorStream.println("Creating Antora playbook ...")
+    Task.log.streams.err.println("Creating Antora playbook ...")
     // dependency to sources
     source()
     val docSite = Task.dest
@@ -360,7 +360,7 @@ object `package` extends RootModule {
       data = githubPagesPlaybookText(authorMode).apply().apply(extraSources),
       createFolders = true
     )
-    Task.log.errorStream.println("Running Antora ...")
+    Task.log.streams.err.println("Running Antora ...")
     runAntora(
       npmDir = npmBase(),
       workDir = docSite,
@@ -376,12 +376,12 @@ object `package` extends RootModule {
     os.write(siteDir / ".nojekyll", "")
 
     // sanitize devAntora source URLs
-    Task.log.errorStream.println("Sanitizing links ...")
+    Task.log.streams.err.println("Sanitizing links ...")
     sanitizeDevUrls(siteDir, devAntoraSources().path, Task.workspace / "docs", Task.workspace)
 
     // only copy the "api" sub-dir; api docs contains a top-level index.html with we don't want
     val unidocSrc = if (authorMode) site.unidocLocal().path else site.unidocSite().path
-    Task.log.errorStream.println(s"Copying API docs from ${unidocSrc} ...")
+    Task.log.streams.err.println(s"Copying API docs from ${unidocSrc} ...")
     os.copy(unidocSrc, siteDir / "api/latest", createFolders = true)
 
     PathRef(siteDir)
