diff --git a/build.sc b/build.sc
index 1e5a27c468..eb655931ab 100644
--- a/build.sc
+++ b/build.sc
@@ -405,10 +405,9 @@ trait BaseMillTestsModule extends TestModule {
 
 /** A MillScalaModule with default set up test module. */
 trait MillAutoTestSetup extends MillScalaModule {
-  // instead of `object test` which can't be overridden, we hand-made a val+class singleton
   /** Default tests module. */
-  val test = new Tests(implicitly)
-  class Tests(ctx0: mill.define.Ctx) extends mill.Module()(ctx0) with super.MillScalaModuleTests
+  lazy val test = new Tests {}
+  trait Tests extends super.MillScalaModuleTests
 }
 
 /** Published module which does not contain strictly handled API. */
@@ -895,8 +894,8 @@ object bsp extends MillModule with BuildInfo {
     )
   }
 
-  override val test = new Test(implicitly)
-  class Test(ctx0: mill.define.Ctx) extends Tests(ctx0) {
+  override lazy val test: Test = new Test {}
+  trait Test extends Tests{
     override def forkEnv: Target[Map[String, String]] = T {
       // We try to fetch this dependency with coursier in the tests
       bsp.worker.publishLocal()()
