diff --git a/build.sc b/build.sc
index 526ebaa9cf..616be491f9 100644
--- a/build.sc
+++ b/build.sc
@@ -1968,7 +1968,7 @@ def uploadToGithub(authKey: String) = task.command {

 private def resolveTasks[T](taskNames: String*): Seq[NamedTask[T]] = {
   mill.resolve.Resolve.Tasks.resolve(
-    build,
+    build.`package`,
     taskNames,
     SelectMode.Separated
   ).map(x => x.asInstanceOf[Seq[mill.define.NamedTask[T]]]).getOrElse(???)