diff --git a/build.mill b/build.mill
index bb60d3b7bee..78ec19509ea 100644
--- a/build.mill
+++ b/build.mill
@@ -10,12 +10,12 @@ import millbuild.*
 //import com.github.lolgab.mill.mima.Mima
 import coursier.maven.MavenRepository
 import coursier.VersionConstraint
-import mill.vcs.VcsVersion
+import mill.util.VcsVersion
 //import com.goyeau.mill.scalafix.ScalafixModule
 import mill._
 import mill.util.Tasks
 import mill.scalalib._
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.api.SelectMode
 import mill.T
diff --git a/contrib/package.mill b/contrib/package.mill
index 38f05bccde9..1cbbd237d33 100644
--- a/contrib/package.mill
+++ b/contrib/package.mill
@@ -5,7 +5,7 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.util.Tasks
 import mill.scalalib._
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.api.SelectMode
diff --git a/example/package.mill b/example/package.mill
index 78f55ecd24c..34caed3efcb 100644
--- a/example/package.mill
+++ b/example/package.mill
@@ -5,7 +5,7 @@ import coursier.maven.MavenRepository
 import mill.*
 import mill.util.Tasks
 import mill.scalalib.*
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.scalalib.publish.*
 import mill.util.Jvm
 import mill.api.SelectMode
diff --git a/integration/package.mill b/integration/package.mill
index 7f1e9cd2894..0040ffa3b5f 100644
--- a/integration/package.mill
+++ b/integration/package.mill
@@ -5,14 +5,14 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.util.Tasks
 import mill.scalalib._
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.api.SelectMode
 import mill.contrib.buildinfo.BuildInfo
 import mill.T
 import mill.api.Cross
-import mill.testrunner.TestResult
+import mill.javalib.testrunner.TestResult
 import millbuild.*
 import upickle.implicits.namedTuples.default.given
 import mill.api.BuildCtx
diff --git a/libs/graphviz/src/mill/main/graphviz/GraphvizTools.scala b/libs/graphviz/src/mill/main/graphviz/GraphvizTools.scala
index d8b6b26d356..832c39a3388 100644
--- a/libs/graphviz/src/mill/main/graphviz/GraphvizTools.scala
+++ b/libs/graphviz/src/mill/main/graphviz/GraphvizTools.scala
@@ -1,4 +1,4 @@
-package mill.main.graphviz
+package mill.graphviz
 
 import com.caoccao.javet.annotations.V8Function
 import com.caoccao.javet.interception.logging.JavetStandardConsoleInterceptor
diff --git a/libs/javalib/package.mill b/libs/javalib/package.mill
index 7b84504be3d..4b5b37db0f4 100644
--- a/libs/javalib/package.mill
+++ b/libs/javalib/package.mill
@@ -6,7 +6,7 @@ import coursier.maven.MavenRepository
 import mill._
 import mill.util.Tasks
 import mill.scalalib._
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.scalalib.publish._
 import mill.util.Jvm
 import mill.api.SelectMode
diff --git a/libs/util/package.mill b/libs/util/package.mill
index 15e100dce96..d245b62d774 100644
--- a/libs/util/package.mill
+++ b/libs/util/package.mill
@@ -6,7 +6,7 @@ import mill.scalalib._
 import mill.contrib.buildinfo.BuildInfo
 import mill.T
 import mill.api.Cross
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import millbuild.*
 
 /**
diff --git a/libs/util/src/mill/util/VisualizeModule.scala b/libs/util/src/mill/util/VisualizeModule.scala
index b752a184106..884554784e5 100644
--- a/libs/util/src/mill/util/VisualizeModule.scala
+++ b/libs/util/src/mill/util/VisualizeModule.scala
@@ -155,7 +155,7 @@ object VisualizeModule extends ExternalModule {
           g = g.graphAttr().`with`(Rank.dir(RankDir.LEFT_TO_RIGHT))
 
           mill.util.Jvm.callProcess(
-            mainClass = "mill.main.graphviz.GraphvizTools",
+            mainClass = "mill.graphviz.GraphvizTools",
             classPath = toolsClasspath().map(_.path).toVector,
             mainArgs = Seq(s"${os.temp(g.toString)};$dest;txt,dot,json,png,svg"),
             stdin = os.Inherit,
diff --git a/mill-build/build.mill b/mill-build/build.mill
index 154569714a3..79609fb5233 100644
--- a/mill-build/build.mill
+++ b/mill-build/build.mill
@@ -10,7 +10,7 @@ object `package` extends MillBuildRootModule {
   override def mvnDeps = Seq(
     // Hardcode MIMA version so test-mill-bootstrap.sh works with
     // the locally-built `SNAPSHOT` version
-    mvn"com.github.lolgab::mill-mima_mill1.0.0-RC3:0.2.0-M4",
+//    mvn"com.github.lolgab::mill-mima_mill1.0.0-RC3:0.2.0-M4",
     mvn"net.sourceforge.htmlcleaner:htmlcleaner:2.29",
     // TODO: implement empty version for ivy deps as we do in import parser
     mvn"com.lihaoyi::mill-contrib-buildinfo:${mill.api.BuildInfo.millVersion}",
diff --git a/mill-build/src/millbuild/Deps.scala b/mill-build/src/millbuild/Deps.scala
index e636a30d4a1..0516ef73815 100644
--- a/mill-build/src/millbuild/Deps.scala
+++ b/mill-build/src/millbuild/Deps.scala
@@ -1,7 +1,7 @@
 package millbuild
 
 import mill.scalalib.*
-import mill.jvmlib.api.*
+import mill.javalib.api.*
 
 object Deps {
 
diff --git a/mill-build/src/millbuild/MillScalaModule.scala b/mill-build/src/millbuild/MillScalaModule.scala
index ceaeee3fc19..195b8c93d42 100644
--- a/mill-build/src/millbuild/MillScalaModule.scala
+++ b/mill-build/src/millbuild/MillScalaModule.scala
@@ -2,7 +2,7 @@ package millbuild
 
 import mill.*
 import mill.scalalib.*
-import mill.jvmlib.api.JvmWorkerUtil
+import mill.javalib.api.JvmWorkerUtil
 import mill.api.BuildCtx
 // import com.goyeau.mill.scalafix.ScalafixModule
 
diff --git a/mill-build/src/millbuild/MillStableScalaModule.scala b/mill-build/src/millbuild/MillStableScalaModule.scala
index ef25e2a891f..11e78faf6da 100644
--- a/mill-build/src/millbuild/MillStableScalaModule.scala
+++ b/mill-build/src/millbuild/MillStableScalaModule.scala
@@ -1,15 +1,15 @@
 package millbuild
-import com.github.lolgab.mill.mima._
+//import com.github.lolgab.mill.mima._
 import mill._, scalalib._
 
 /** Publishable module which contains strictly handled API. */
-trait MillStableScalaModule extends MillPublishScalaModule with Mima {
+trait MillStableScalaModule extends MillPublishScalaModule /*with Mima*/ {
 
-  override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq.empty[ProblemFilter]
-
-  def mimaPreviousVersions: T[Seq[String]] = Settings.mimaBaseVersions
-
-  def mimaPreviousArtifacts: T[Seq[Dep]] = Task {
+//  override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq.empty[ProblemFilter]
+//
+//  def mimaPreviousVersions: T[Seq[String]] = Settings.mimaBaseVersions
+//
+//  def mimaPreviousArtifacts: T[Seq[Dep]] = Task {
 //    Settings.mimaBaseVersions
 //      .map({ version =>
 //        val patchedSuffix = {
@@ -27,8 +27,8 @@ trait MillStableScalaModule extends MillPublishScalaModule with Mima {
 //        val patchedId = artifactName() + patchedSuffix
 //        mvn"${pomSettings().organization}:${patchedId}:${version}"
 //      })
-    Seq.empty[Dep]
-  }
-
-  def mimaExcludeAnnotations = Seq("mill.api.internal.internal", "mill.api.experimental")
+//    Seq.empty[Dep]
+//  }
+//
+//  def mimaExcludeAnnotations = Seq("mill.api.internal.internal", "mill.api.experimental")
 }
diff --git a/website/package.mill b/website/package.mill
index dd0965f8778..ee1476b87fa 100644
--- a/website/package.mill
+++ b/website/package.mill
@@ -105,7 +105,7 @@ object `package` extends mill.Module {
 
     expandDiagramsInDirectoryAdocFile(
       Task.dest,
-      mill.main.VisualizeModule.toolsClasspath().map(_.path)
+      mill.util.VisualizeModule.toolsClasspath().map(_.path)
     )
 
     PathRef(Task.dest)
@@ -171,7 +171,7 @@ object `package` extends mill.Module {
     }
 
     mill.util.Jvm.callProcess(
-      mainClass = "mill.main.graphviz.GraphvizTools",
+      mainClass = "mill.graphviz.GraphvizTools",
       classPath = visualizeClassPath.toSeq,
       mainArgs = orderedDiagrams.map { case (p, i, src, dest) =>
         Task.log.debug(s"Rendering graphviz: ${p} (${i}) to ${dest}")
@@ -233,7 +233,7 @@ object `package` extends mill.Module {
     os.copy(blogFolder0().path, Task.dest, mergeFolders = true)
     expandDiagramsInDirectoryAdocFile(
       Task.dest,
-      mill.main.VisualizeModule.toolsClasspath().map(_.path)
+      mill.util.VisualizeModule.toolsClasspath().map(_.path)
     )
     PathRef(Task.dest)
   }
@@ -352,7 +352,7 @@ object `package` extends mill.Module {
 
       expandDiagramsInDirectoryAdocFile(
         outputFolder,
-        mill.main.VisualizeModule.toolsClasspath().map(_.path)
+        mill.util.VisualizeModule.toolsClasspath().map(_.path)
       )
 
       val useOldDownloadUrl =
diff --git a/dist/package.mill b/dist/package.mill
index fc069e6357c..d999c1fef95 100644
--- a/dist/package.mill
+++ b/dist/package.mill
@@ -89,9 +89,9 @@ trait DistModule extends Module {
   def publishArtifacts = Task {
     PublishModule.PublishData(
       meta = artifact(),
-      payload = Seq(
-        executableRaw() -> s"${artifactFileNamePrefix()}.exe",
-        pom() -> s"${artifactFileNamePrefix()}.pom"
+      payload = Map(
+        os.SubPath(s"${artifactFileNamePrefix()}.exe") -> executableRaw(),
+        os.SubPath(s"${artifactFileNamePrefix()}.pom") -> pom()
       )
     )
   }
@@ -258,10 +258,10 @@ object `package` extends MillJavaModule with DistModule {
   def publishArtifacts = Task {
     super.publishArtifacts().copy(payload =
       super.publishArtifacts().payload ++
-        exampleZips().map(z => (z, z.path.last)) ++
-        Seq(
-          (bootstrapLauncher(), s"${artifactFileNamePrefix()}-mill.sh"),
-          (bootstrapLauncherBat(), s"${artifactFileNamePrefix()}-mill.bat")
+        exampleZips().map(z => os.SubPath(z.path.last) -> z) ++
+        Map(
+          os.SubPath(s"${artifactFileNamePrefix()}-mill.sh") -> bootstrapLauncher(),
+          os.SubPath(s"${artifactFileNamePrefix()}-mill.bat") -> bootstrapLauncherBat()
         )
     )
   }
